src/server_manager/electron_app/gcp_oauth.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as electron from 'electron';
+import * as express from 'express';
+import {OAuth2Client} from 'google-auth-library';
+import {AddressInfo} from 'net';
+
+const OAUTH_CONFIG = {
+  // TODO: Create GCP project under firehook-products
+  project_id: 'mpmcroy-server-provisioner',
+  client_id: '276807089705-e6sk8e96a2kbuilgnehfaag75ab2aom3.apps.googleusercontent.com',
+  scopes: [
+    'https://www.googleapis.com/auth/userinfo.email',
+    'https://www.googleapis.com/auth/cloud-platform',
+    'https://www.googleapis.com/auth/compute',
+    'https://www.googleapis.com/auth/devstorage.full_control',
+  ],
+};
+const REDIRECT_PATH = '/gcp/oauth/callback';
+
+function responseHtml(messageHtml: string): string {
+  return `<html><script>window.close()</script><body>${
+      messageHtml}. You can close this window.</body></html>`;
```

Comment:
We should improve this at some point. It's not even internationalized. But it's ok for now.

Reply:
Sounds good